temba/channels/views.py

```
@@ -1844,6 +1846,59 @@ def form_valid(self, form):
+        class LineForm(forms.Form):
+            channel_id = forms.CharField(label=_("Channel ID"), required=True, help_text=_("The ID of the your Channel on Business LINE"))
+            channel_secret = forms.CharField(label=_("Channel Secret"), required=True, help_text=_("The Secret of the your Channel on Business LINE"))
+            channel_mid = forms.CharField(label=_("Channel MID"), required=True, help_text=_("The MID of the your Channel on Business LINE"))
+
+            def clean(self):
+                from linebot.client import LineBotClient
+
+                channel_id = self.cleaned_data.get('channel_id')
+                channel_secret = self.cleaned_data.get('channel_secret')
+                channel_mid = self.cleaned_data.get('channel_mid')
+
+                existing = Channel.objects.filter(channel_type=Channel.TYPE_LINE, address=channel_mid, is_active=True).first()
+                if existing:
+                    raise ValidationError(_("Channel with this address already exists, please try again with other."))
+
+                credentials = {
+                    'channel_id': channel_id,
+                    'channel_secret': channel_secret,
+                    'channel_mid': channel_mid
+                }
+
+                line_bot_client = LineBotClient(**credentials)
+                try:
+                    users = line_bot_client.get_user_profile(channel_mid)
+                    profile = [
+                        {'mid': user._UserProfile__profile.get('mid'),
+                         'picture_url': user._UserProfile__profile.get('picture_url'),
+                         'display_name': user._UserProfile__profile.get('display_name'),
+                         'status_message': user._UserProfile__profile.get('status_message')} for user in users][0]
+                except:
+                    profile = {'mid': channel_mid, 'picture_url': None, 'display_name': channel_id, 'status_message': None}
+
```

Why continue if we fail to get the bot info? This should raise an exception.